File: day_6/Find_All_Duplicates_in_an_Array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        dup = []
        for num in nums:
            num = abs(num)
            nums[num - 1] *= -1
            if nums[num - 1] > 0:
                dup.append(num)
        return dup

File: day_6/Find_All_Duplicates_in_an_Array.js
/** Given an array of integers, 1 ≤ a[i] ≤ n (n = size of array), some elements appear twice and others appear   once. Find all the elements that appear twice in this array without extra space and in O(n) runtime.
 * @param {number[]} nums
 * @return {number[]}
 */
let findDuplicates = function(nums) {
    let dup = [],
        ind;
    for (let num of nums){
        ind = Math.abs(num) - 1;
        nums[ind] *= -1;
        if (nums[ind] > 0){
            dup.push(ind + 1);
        }
    }
    return dup;
};